.github/workflows/gradle.yml

```diff
@@ -43,6 +65,19 @@ jobs:
         with:
           python-version: '3.7'
 
+      - name: Start local Docker registry
+        run: docker run -d -p 5000:5000 --restart=always --name registry -v /tmp/docker-registry:/var/lib/registry registry:2 && npx wait-on tcp:5000
+
+      - name: Build
+        run: ./gradlew --no-daemon build --scan
+
+      - name: Ensure no file change
+        run: git status --porcelain && test -z "$(git status --porcelain)"
+
+      - name: Check documentation
+        if: success() && github.ref == 'refs/heads/master'
+        run: ./tools/site/link_checker.sh check_docs
+
       - name: Write Integration Test Credentials
```

Comment:
I moved this block down to make sure that `./gradlew build` doesn't accidentally include a step that depends on credentials.